app/controllers/admin/users_controller.rb

```diff
@@ -514,6 +515,29 @@ def sso_record
     render json: success_json
   end
 
+  def delete_associated_accounts
+    guardian.ensure_can_delete_user_associated_accounts!(@user)
+    previous_value =
+      @user
+        .user_associated_accounts
+        .select(:provider_name, :provider_uid, :info)
+        .map do |associated_account|
+          {
+            provider: associated_account.provider_name,
+            uid: associated_account.provider_uid,
+            info: associated_account.info,
+          }.to_s
+        end
+        .join(",")
+    StaffActionLogger.new(current_user).log_delele_associated_accounts(
+      @user,
+      previous_value:,
+      context: params[:context],
+    )
+    @user.user_associated_accounts.destroy_all
```

Comment:
If we use `destroy` there's a chance that it will fail, but we seem to rely on this always succeeding, so we should probably use `#delete_all` instead. 🤔 

But we can only do that if we don't need any callbacks to be invoked. If we need it to be invoked, we should probably staff-action-log and render conditionally based on the result of `destroy_all`.